### PR TITLE
Add Alpine info, fix headers, restructure content

### PR DESCRIPTION
**Pull request by jyee**
https://trello.com/c/jsOPC0qa/1212-add-mention-of-alpine-builds-in-docker-docs

Header levels didn't match up, so I updated them. At the same time it made sense to re-arrange some of the content so it would be more organized.

### CODE REVIEW

**Comment by irabinovitch** [comment]
LGTM merge away.

File: content/integrations/docker.md [hunk 1]
```diff
@@ -12,56 +12,99 @@ git_integration_title: docker
...
 
 ### Environment variables
 
-A few parameters can be changed with environment variables.
+Note that in the command above, you are able to pass your API key to the Datadog Agent using Docker's `-e` or environment variable flag. Some other variables you can pass include:
+
+* **TAGS** sets host tags. You can pass both simple tags and key-value tags. For example, `-e TAGS="simple-tag,tag-key:tag-value"`.
+* **LOG_LEVEL** sets logging verbosity (CRITICAL, ERROR, WARNING, INFO, DEBUG). For example, `-e LOG_LEVEL=DEBUG` will set logging to debug mode.
+* **PROXY_HOST**, **PROXY_PORT**, **PROXY_USER**, and **PROXY_PASSWORD** set the proxy configuration.
+* **DD_URL** sets the Datadog intake server where the agent will send metrics (used when using an agent as a proxy).
```
Comment (jyee) [inline content/integrations/docker.md]: Thanks. Yeah, I think we should add the SD stuff in passing and point to the SD docs. I'll also add the other env vars.

**Comment by jyee** [comment]
@irabinovitch @technovangelist One quick review of the env var table I added before merging?

**Comment by irabinovitch** [comment]
LGTM merge away.

**Pull request merged**